object/object.go

```diff
@@ -23,23 +23,50 @@ const (
 	ARRAY_OBJ    = "ARRAY"
 )
 
+type TraceMetadata struct {
+	Dependencies map[int]bool
+}
+
+func MergeDependencies(a TraceMetadata, b TraceMetadata) TraceMetadata {
```

Comment:
This algorithm is a bit... brute force. I'm not too worried about it but if it becomes a bottleneck I'm down to brainstorm alternatives